federation/executor.go

```diff
@@ -280,7 +292,7 @@ func (e *Executor) execute(ctx context.Context, p *Plan, keys []interface{}) (in
 					deleteKey(v, federationField)
 					subPlanMetaData.results[k] = v
 				} else {
-					if k != keyField || v != subPlanMetaData.results[k] {
+					if k != keyField && v == subPlanMetaData.results[k] {
```

Comment:
Why to make this change?

Reply:
Ah, that was a mistake due to tests, nice catch!